### PR TITLE
Fix docker setup

### PR DESCRIPTION
**Pull request by roschaefer**
_x_

### CODE REVIEW

**Comment by roschaefer** [comment]
Ok, so there is a slight hack here. The docker image must have the right `.env` at **build** time. So, I added all environment variables which are *public* to `.env.staging` and `.env.example` respectively. We still need `.env.production`. During build, depending on build argument `ENV_FILE`, the desired config file is moved into place. All *secret* environment variables can be passed as environment variables at **runtime** and are not be exposed to the client. As long as we have no secrets that need to be available at **build** time, this approach should work.

**Comment by appinteractive** [comment]
So you say that we are not able anymore to deploy that container outside of the fixed environments we specify on build? And we are not able to specify vars (which would be „variable“) at runtime anymore? 

That point is critical as we have to be able to deploy fast with changes settings without waiting for a whole build every time. 

The standard behavior for docker images should be always, build once, deploy everywhere. We really need to keep that workflow intact.

File: Dockerfile [hunk 1]
```diff
@@ -1,46 +1,32 @@
...
+# optional git commit hash
+ARG BUILD_COMMIT
+ENV BUILD_COMMIT=$BUILD_COMMIT
 
-ENTRYPOINT ["./entrypoint.sh"]
+RUN mkdir -p /WebApp/
+WORKDIR /WebApp/
+# --no-cache: download package index on-the-fly, no need to cleanup afterwards
+# --virtual: bundle packages, remove whole bundle at once, when done
+RUN apk --no-cache --virtual build-dependencies add git python make g++
```
[the remark this answers is not in the document]
Comment (appinteractive) [reply Dockerfile]: with node:8-alpine it does work, with node:alpine not

File: plugins/raven-server.js [hunk 2]
```diff
@@ -1,19 +1,14 @@
 import Raven from 'raven'
-import dotenv from 'dotenv'
-import {readFileSync} from 'fs'
-import {resolve} from 'path'
 
 export default async function (app) {
-  const secrets = dotenv.parse(await readFileSync(resolve('./server/.env-secrets')))
-
-  if (!process.client && secrets.SENTRY_DNS_PRIVATE) {
+  if (!process.client && process.env.SENTRY_DNS_PRIVATE) {
```
[the remark this answers is not in the document]
Comment (roschaefer) [reply plugins/raven-server.js]: This file is executed on the server. I double checked at the time, that `console.log(process.env.WHATEVER)` spits out the `WHATEVER` variable server-side but not client-side.

**Comment by appinteractive** [comment]
@roschaefer how was your procedure for testing if the env vars work correctly?

File: plugins/raven-server.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (roschaefer) [reply plugins/raven-server.js]: If we had env variables that we need on the client, we would have to explicitly whitelist them.

File: nuxt.config.js [hunk 3]
```diff
@@ -112,8 +111,20 @@ module.exports = {
     {src: '~/plugins/open-page-in-modal.js', ssr: false}
   ],
   modules: [
-    'cookie-universal-nuxt',
-    '@nuxtjs/dotenv'
+    ['@nuxtjs/dotenv', {
```
[the remark this answers is not in the document]
Comment (roschaefer) [reply nuxt.config.js]: ? The config file name? I was playing with if statements in here based on the current `process.env.NODE_ENV` when nuxt reads in this file but that never fully worked as it should. So I went back to `env-sub`.

**Comment by appinteractive** [comment]
Just as a reference after we had a talk together: the goal is to not build the application every time the docker image is started but only run it while at the same time providing current environemnt variables that might be passed to the docker command to the application (without rebuild)

**Comment by roschaefer** [comment]
@appinteractive changing the app via env variables should be possible now.

**Comment by appinteractive** [comment]
```
docker build -t humanconnection/frontend-nuxt .
```

Still getting the same issue: 
```
Step 13/17 : RUN yarn install --production=false --frozen-lockfile --non-interactive
 ---> Running in 05485f872598
yarn install v0.23.2
[1/4] Resolving packages...
[2/4] Fetching packages...
error nuxt@1.4.0: The engine "node" is incompatible with this module. Expected version ">=8.0.0".
error Found incompatible module
```

Changing the minimum node version in the package.json does not help
```
"engines": {
    "node": ">= 10.0.0",
    "yarn": ">= 1.3.0"
  },
```

**Comment by appinteractive** [comment]
I fixed the issue by pinning the node version to the future Node 10.* LTS.

@roschaefer how about the env issues? Any updates there? Sounded like there are still not resolved!?

**Comment by roschaefer** [comment]
@appinteractive I didn't get any build issues at the time, my last commit also greened the build server. Glad if it is fixed on more than one machine now :laughing: . This branch here is OK :ok_hand: the env issues are a problem in #228 but not here.

**Comment by appinteractive** [comment]
@roschaefer I still could not test it as I can’t access the docker image via browser. Any suggestions how to test if the bars are really there?

**Comment by Lulalaby** [comment]
@appinteractive & @roschaefer Ready for #233 ?

**Comment by appinteractive** [comment]
Can test it as soon as Monday. Staging is down until Monday too. Would like to first get it working to see if this merge brakes something.

**Comment by Lulalaby** [comment]
Okay

File: .travis.yml [hunk 4]
```diff
@@ -1,25 +1,26 @@
...
   yarn: true
   directories:
     - node_modules
 
-install:
-  # nothing
+# install current version of yarn
+before_install:
+  - curl -o- -L https://yarnpkg.com/install.sh | bash
+  - export PATH="$HOME/.yarn/bin:$PATH"
```
Comment (roschaefer) [inline .travis.yml]: @appinteractive is this redundant?

Comment (appinteractive) [reply .travis.yml]: Yes it can be removed. Was caused by the unversionated bode image before.

**Pull request merged**